patch-package
new file mode 100644
Binary files /dev/null and b/node_modules/web3-providers-ws/.DS_Store differ
--- a/node_modules/web3-providers-ws/src/index.js
+++ b/node_modules/web3-providers-ws/src/index.js
@@ -29,11 +29,24 @@ var Ws = null;
 var _btoa = null;
 var parseURL = null;
 if (typeof window !== 'undefined') {
-    Ws = window.WebSocket;
-    _btoa = btoa;
-    parseURL = function(url) {
-        return new URL(url);
+    if (window.hasOwnProperty('WebSocket'))
+        Ws = window.WebSocket;
+    else
+        Ws = require('websocket').w3cwebsocket;
+
+    if (window.hasOwnProperty('btoa'))
+        _btoa = btoa;
+    else
+        _btoa = function(str) {
+      return Buffer(str).toString('base64');
     };
+
+    if (window.hasOwnProperty('URL'))
+        parseURL = function(url) {
+            return new URL(url);
+        };
+    else
+        parseURL = require('url').parse;
 } else {
     Ws = require('websocket').w3cwebsocket;
     _btoa = function(str) {
